docs/design/queue-resource-reservation-design.md

```diff
@@ -0,0 +1,110 @@
+* Take preemption and non-preemption reservation into consideration.
+* Preemption reservation may results in running workloads exits abnormally.
+* Non-preemption reservation needs gentle treatment to balance scheduling and reserving performance.
+* Although preemption reservation configured, try to reserve resource without eviction if idle resource can satisfy demand.
+* Nodes locked by target job cannot be chosen as locked node.
+* Nodes locked by another queue cannot be chosen as locked node.
+* Try to lock nodes, whose total resource is as less as possible on base of satisfying requirement, to avoid dramatic 
+scheduling performance degradation.
+
+### Safety
+* Malicious application for large amount of resource reservation will cause jobs in other queue to block.
+* Preemption reservation has a risk of hurting existing business.
+
+## Design
+### API
+```
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: q1
+spec:
+  reclaimable: true
+  weight: 1
+
+  guarantee:            // reservation key word
+    policy: Best-Effort // preemption reservation or non-preemption reservation
+    percentage: 0.5     // locked nodes number percentage in cluster
+    resource:           // specified reserving resource
+      cpu: 2c
+      memory: 4G
+
+status:
+  state: Open
+
+  reservation:          // reservation status key word
+    nodes:              // locked nodes list
+      - n1
+      - n2
+    resource:           // total idle resource in locked nodes
+      cpu: 1C
+      memory: 2G 
+```
+### Detail
+#### Fields
+##### policy
+Option Field, options are "Best-Effort" and "Guaranteed" and "Best-Effort" by default. If "Best-Effort" is set, scheduler
+will reserve resource without evicting running workloads, while "Guaranteed" is opposite.
+#### percentage
+Option Field if `resource` not specified, legal value range is [0, 1]. Locked node number will be as follows: 
+```
+math.Floor(clusterNodeNumber * percentage)
+```
+It should be noted that nodes will be locked randomly util reaching the percentage. If `resource` is also specified, choose
+the one with more reserved resources as the final result.
+#### guarantee.resource
+Option Field if `percentage` not specified. List of reserving resource categories and amount. 
+#### nodes
+List of locked nodes.
+#### status.resource
+Total idle resource of locked nodes. You can judge whether reservation has met demand by watching this field.
+
+#### Algorithm
+##### Node Selection
+* Big Nodes
+
+Sort all nodes according total resource amount of one or some dimensions. Select the least nodes which rank first and 
+satisfy requirement. It's the most efficient way for preemption reservation.  
+
+* Idle First
+
+Sort all nodes according idle resource of one or some dimensions. Select the least nodes which rank first as locked nodes.
+It's suitable for non-preemption. It's should to be noted that the locked nodes are often optimal solution but suboptimal
+solution sometimes. 
+
```

Comment:
add another algorithm:  sort by existing pods belonging to target queue in descending order. select nodes which can meet the reservation requirement.